### DISCUSSION TITLE
Handle resource removal

### DISCUSSION THREAD

**Discussion by eliecharra** [spacelift-io/spacelift-operator#51] [Ideas]
Spacelift resources created through the operator are currently not removed when you remove them from the cluster.
This breaks a bit the purpose of the operator and leads to the generation of orphaned resources.

We can implement finalizers on resources, and hook into the deletion process to remove them from Spacelift when they are removed from the cluster.

**Open question**

If we allow [management of existing resources](https://github.com/spacelift-io/spacelift-operator/discussions/50) with the operator, should we also remove those on Spacelift?

**Comment by gaetansnl** [comment]
Hello @eliecharra do you know if there is an ETA for this issue ?

**Reply by eliecharra** [reply]
Hey @gaetansnl, we don't have any ETA so far. 

We do have other higher priority topics in our backlog.
Since this operator is not getting that much traction, it's not currently in our top list.

That being said, we're happy to review contributions and merge third party PRs here.
That being said, working on resource removal will require us to agree on the technical approach first. One that could work for any kind of resource managed here.
